src/t_zset.c

```diff
@@ -115,15 +115,40 @@ static inline void zslSetNodeHeight(zskiplistNode *x, int height) {
 }
 
 /* Create a skiplist node with the specified number of levels.
- * The SDS string 'ele' is referenced by the node after the call. */
-static zskiplistNode *zslCreateNode(int height, double score, sds ele) {
-    zskiplistNode *zn = zmalloc(sizeof(*zn) + height * sizeof(struct zskiplistLevel));
+ * By embedding elements and levels into the skiplist nodes,
+ * we achieve better cache-friendliness and a more compact memory structure.
```

Comment:
Better and more compact than what? Once this is merged, the old representation is history.

```suggestion
 * we achieve good cache-friendliness and a compact memory structure.
```